/*
 * Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
 */

import { z } from "zod";

export type CALLEXCEPTIONContext = {};

export type CallException = {
    context?: CALLEXCEPTIONContext | undefined;
    errorCode?: string | undefined;
    message?: string | undefined;
    path: string;
    statusCode?: number | undefined;
    timestamp: string;
};

/** @internal */
export namespace CALLEXCEPTIONContext$ {
    export type Inbound = {};

    export const inboundSchema: z.ZodType<CALLEXCEPTIONContext, z.ZodTypeDef, Inbound> = z.object(
        {}
    );

    export type Outbound = {};

    export const outboundSchema: z.ZodType<Outbound, z.ZodTypeDef, CALLEXCEPTIONContext> = z.object(
        {}
    );
}

/** @internal */
export namespace CallException$ {
    export type Inbound = {
        context?: CALLEXCEPTIONContext$.Inbound | undefined;
        errorCode?: string | undefined;
        message?: string | undefined;
        path: string;
        statusCode?: number | undefined;
        timestamp: string;
    };

    export const inboundSchema: z.ZodType<CallException, z.ZodTypeDef, Inbound> = z
        .object({
            context: z.lazy(() => CALLEXCEPTIONContext$.inboundSchema).optional(),
            errorCode: z.string().default("CALL_EXCEPTION"),
            message: z.string().default("Your call to the blockchain encounters revert."),
            path: z.string(),
            statusCode: z.number().default(400),
            timestamp: z.string(),
        })
        .transform((v) => {
            return {
                ...(v.context === undefined ? null : { context: v.context }),
                errorCode: v.errorCode,
                message: v.message,
                path: v.path,
                statusCode: v.statusCode,
                timestamp: v.timestamp,
            };
        });

    export type Outbound = {
        context?: CALLEXCEPTIONContext$.Outbound | undefined;
        errorCode: string;
        message: string;
        path: string;
        statusCode: number;
        timestamp: string;
    };

    export const outboundSchema: z.ZodType<Outbound, z.ZodTypeDef, CallException> = z
        .object({
            context: z.lazy(() => CALLEXCEPTIONContext$.outboundSchema).optional(),
            errorCode: z.string().default("CALL_EXCEPTION"),
            message: z.string().default("Your call to the blockchain encounters revert."),
            path: z.string(),
            statusCode: z.number().default(400),
            timestamp: z.string(),
        })
        .transform((v) => {
            return {
                ...(v.context === undefined ? null : { context: v.context }),
                errorCode: v.errorCode,
                message: v.message,
                path: v.path,
                statusCode: v.statusCode,
                timestamp: v.timestamp,
            };
        });
}
